File: get_ids.py
import requests
import json
import os

url_endpoint = 'https://archive.org/services/search/v1/scrape'
parms = {'debug':'false','xvar':'production','total_only':'false','count':'10000','sorts':'date asc,avg_rating desc,num_favorites desc,downloads desc','fields':'identifier,date,avg_rating,num_reviews,num_favorites,stars,downloads,files_count,format,collection,source,subject,type'}

def get_ids(year,parms=parms,prev_rows=0):
  query = 'collection:GratefulDead AND year:'+str(year)
  parms['q'] = query
  r = requests.get(url_endpoint,params=parms)
  print("url is {}".format(r.url))
  if r.status_code != 200: print ("error collecting year {}".format(year)); raise Exception('Download','Error {} collection year {}'.format(r.status_code,year))
  j = r.json()
  current_rows = prev_rows + j['count']
  if current_rows != j['total']:  
    print ("there are {} some missing rows in year {}".format(j['total']-current_rows,year))
  #  parms['cursor'] = j['cursor']
  #  parms.pop('q')
  #  get_ids(year,parms,prev_rows=current_rows) 
  return r.json()

for year in range(1965,1996,1):
  ids = get_ids(year) 
  pathname = os.path.join(os.getenv('HOME'),'projects','dead_vault','data',str(year),'ids.json')
  os.makedirs(os.path.dirname(pathname),exist_ok=True)
  json.dump(ids,open(pathname,'w'))

File: get_ids.js

const fs = require('fs');
const https = require('https');

let start=1966;
let end = 1966;

total_result = {};
function id_promise(year){
let URL = 'https://archive.org/services/search/v1/scrape?debug=false&xvar=production&total_only=false&count=1000&fields=identifier%2Cdate%2Cnum_reviews%2Cnum_favorites%2Cfiles_count%2Cavg_rating&q=collection%3AGratefulDead AND year%3A' + year;
  return new Promise((resolve,reject) => {
    console.log ("URL is "+URL);
    https.get(URL,(resp)=> {
      let data = [];
      resp.on('data',(chunk)=> { data.push(chunk);});
      resp.on('end',()=> { 
         console.log("Total Items Fetched: " + JSON.parse(data).total); 
         total_result[year] = Buffer.concat(data).toString();
         resolve(total_result[year]);
      });
     }).on("error",(err)=> {reject(err);});
 });
}

function new_get_ids(year){
  let resp = id_promise(year).then(
     result => console.log("Success "+year),
     error => console.log(error));
}

function request_identifiers(year){
  let URL = 'https://archive.org/services/search/v1/scrape?debug=false&xvar=production&total_only=false&count=1000&fields=identifier%2Cdate%2Cnum_reviews%2Cnum_favorites%2Cfiles_count%2Cavg_rating&q=collection%3AGratefulDead AND year%3A' + year;
  console.log ("URL is "+URL);
  let data = '';
  https.get(URL,(resp)=> {
    resp.on('data',(chunk)=> { data += chunk;});
    resp.on('end',()=> { 
       console.log("Total Items Fetched: " + JSON.parse(data).total); 
       return data});
   }).on("error",(err)=> {console.log("Error: "+err.message);});
}
/*
let id_dict = {};
for (let year=start; year<end+1; year++){
  console.log ("year is " + year);
  id_dict[year] = request_identifiers(year);
};

fs.writeFile('/home/steve/projects/dead_vault/data',JSON.stringify(id_dict));
*/
